File: CodeForces/TheForces Round #17 (AOE-Forces)/E.cpp
#include <iostream>
#include <vector>

const int MOD = 1e9 + 7;

int countOnes(int n) {
    int count = 0;
    while (n) {
        count++;
        n &= n - 1;
    }
    return count;
}

int generateCombinations(int l1, int r1, int l2, int r2, int l3, int r3) {
    int count = 0;
    for (int i = l1; i <= r1; i++) {
        for (int j = l2; j <= r2; j++) {
            for (int k = l3; k <= r3; k++) {
                int xorVal = i ^ j ^ k;
                if (countOnes(xorVal) == 2 || countOnes(xorVal) == 3 ||
                    countOnes(xorVal) == 5 || countOnes(xorVal) == 7 ||
                    countOnes(xorVal) == 11 || countOnes(xorVal) == 13 ||
                    countOnes(xorVal) == 17 || countOnes(xorVal) == 19 ||
                    countOnes(xorVal) == 23 || countOnes(xorVal) == 29 ||
                    countOnes(xorVal) == 31) {
                    count = (count + 1) % MOD;
                }
            }
        }
    }
    return count;
}

std::vector<int> countPrimeXOR(int t, const std::vector<std::vector<std::pair<int, int>>>& testCases) {
    std::vector<int> results;
    for (int i = 0; i < t; i++) {
        int l1 = testCases[i][0].first;
        int r1 = testCases[i][0].second;
        int l2 = testCases[i][1].first;
        int r2 = testCases[i][1].second;
        int l3 = testCases[i][2].first;
        int r3 = testCases[i][2].second;
        int result = generateCombinations(l1, r1, l2, r2, l3, r3);
        results.push_back(result);
    }
    return results;
}

int main() {
    int t;
    std::cin >> t;

    std::vector<std::vector<std::pair<int, int>>> testCases(t);
    for (int i = 0; i < t; i++) {
        testCases[i].resize(3);
        for (int j = 0; j < 3; j++) {
            std::cin >> testCases[i][j].first >> testCases[i][j].second;
        }
    }

    std::vector<int> results = countPrimeXOR(t, testCases);

    for (int i = 0; i < t; i++) {
        std::cout << results[i] << std::endl;
    }

    return 0;
}

File: CodeForces/TheForces Round #17 (AOE-Forces)/E.py
MOD = int(1e9) + 7

def count_prime_xor(t, test_cases):
    def count_ones(n):
        count = 0
        while n:
            count += 1
            n &= n - 1
        return count

    def generate_combinations(l1, r1, l2, r2, l3, r3):
        count = 0
        for i in range(l1, r1 + 1):
            for j in range(l2, r2 + 1):
                for k in range(l3, r3 + 1):
                    xor = i ^ j ^ k
                    if count_ones(xor) in [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31]:
                        count = (count + 1) % MOD
        return count

    results = []
    for i in range(t):
        l1, r1 = test_cases[i][0]
        l2, r2 = test_cases[i][1]
        l3, r3 = test_cases[i][2]
        result = generate_combinations(l1, r1, l2, r2, l3, r3)
        results.append(result)
    return results

# Reading input
t = int(input())
test_cases = []
for _ in range(t):
    armies = [list(map(int, input().split())) for _ in range(3)]
    test_cases.append(armies)

# Counting prime XOR combinations
results = count_prime_xor(t, test_cases)

# Printing results
for result in results:
    print(result)
